src/Http/RouteConfig.php

```diff
@@ -8,18 +8,55 @@
 
 final class RouteConfig
 {
+    /** @var string The mark to give the next route in the matching Regex */
+    private string $regexMark = 'a';
+
     public function __construct(
         /** @var array<string, array<string, \Tempest\Http\Route>> */
-        public array $routes = [],
+        public array  $staticRoutes = [],
+        /** @var array<string, array<string, \Tempest\Http\Route>> */
+        public array  $dynamicRoutes = [],
+        /** @var array<string, string> */
+        public array $matchingRegexes = []
     ) {
     }
 
     public function addRoute(ReflectionMethod $handler, Route $route): self
     {
         $route->setHandler($handler);
 
-        $this->routes[$route->method->value][$route->uri] = $route;
+        if ($route->isDynamic) {
+            $routeMark = $this->regexMark++;
```

Comment:
Out of curiosity, what happens when you reach `z`?

Reply:
It will generate "aa", but researching this, I found out it is soft deprecated in PHP8.3.0! https://www.php.net/manual/en/language.operators.increment.php#language.operators.increment.string

So I replaced it with str_increment.